mypy/nodes.py

```diff
@@ -2334,6 +2341,11 @@ class SymbolTableNode:
     normalized = False  # type: bool
     # Was this defined by assignment to self attribute?
     implicit = False  # type: bool
+    # Is this node refers to other node via node aliasing?
+    # (This is currently used for simple aliases like `A = int` instead of .type_override)
+    is_aliasing = False  # type: bool
+    # This includes full names of aliases used in this alias.
+    alias_depends_on = None  # type: Set[str]
```

Comment:
I wonder if this actually is necessary. Even without this, the target that contains the alias definition should depend on these aliases, right? And change in the alias should be detected by `mypy.server.astdiff`. If you remove this, what will break? Maybe some symbol table differences won't then be propagated correctly -- but that could just be a bug elsewhere, which this papers over.

Reply:
Ideally this shouldn't be necessary. But IIRC correctly some tests failed without this. I added a TODO about this in `typeanal.py` (that you also commented). I will try to remove this and re-run the tests, and if some will still fail, I will try to figure out where exactly is the problem in diff calculation and expand the TODO.